scripts/irods/test/test_ichksum.py

```diff
@@ -466,3 +467,81 @@ def test_ichksum_detects_when_the_size_in_storage_is_less_than_the_size_in_catal
         # Show that ichksum detects the size inconsistency between the catalog and the object in storage.
         self.admin.assert_icommand(['ichksum', '-f', data_object], 'STDERR', ['-512000 UNIX_FILE_READ_ERR'])
 
+    def test_ichksum_server_honors_server_config_option_checksum_read_buffer_size_in_bytes__issue_7947(self):
+
+        def get_checksum(data_object, replica_number, expected_length_of_checksum=49):
+            gql = "select DATA_CHECKSUM where COLL_NAME = '{0}' and DATA_NAME = '{1}' and DATA_REPL_NUM = '{2}'"
+            checksum, err, ec = self.admin.run_icommand(['iquest', '%s', gql.format(self.admin.session_collection, data_object, replica_number)])
+            self.assertEqual(ec, 0)
+            self.assertEqual(len(err), 0)
+            self.assertEqual(len(checksum.strip()), expected_length_of_checksum)
+            return checksum.strip()
```

Comment:
FWIW, we have a function called `get_replica_checksum` in lib: https://github.com/irods/irods/blob/bc2a9bf0d8d3587d1aad8bf6a9b3db650e0d6944/scripts/irods/lib.py#L651-L654 Unfortunately, it only takes a `DATA_NAME` for some reason but that shouldn't be a problem for your particular data object. If we use that, we won't need so many copies of the same function.

Reply:
I made this change for both test_ichksum.py and test_irepl.py.  I left test_iput.py alone since it already had that method as a class variable.